nhrpd/netlink_arp.c

```diff
@@ -265,47 +149,64 @@ void netlink_set_nflog_group(int nlgroup)
 	}
 }
 
-void netlink_init(void)
+void nhrp_neighbor_operation(ZAPI_CALLBACK_ARGS)
 {
-	netlink_req_fd = znl_open(NETLINK_ROUTE, 0);
-	if (netlink_req_fd < 0)
-		return;
+	union sockunion addr = {}, lladdr = {};
+	struct interface *ifp;
+	int state, ndm_state;
+	struct nhrp_cache *c;
+	struct zapi_neigh_ip api = {};
 
-	netlink_listen_fd = znl_open(NETLINK_ROUTE, RTMGRP_NEIGH);
-	if (netlink_listen_fd < 0)
+	zclient_neigh_ip_decode(zclient->ibuf, &api);
+	if (api.ip_in.ipa_type == AF_UNSPEC)
 		return;
+	sockunion_family(&addr) = api.ip_in.ipa_type;
+	memcpy((uint8_t *)sockunion_get_addr(&addr), &api.ip_in.ip.addr,
+	       family2addrsize(api.ip_in.ipa_type));
 
-	thread_add_read(master, netlink_route_recv, 0, netlink_listen_fd, NULL);
-}
+	sockunion_family(&lladdr) = api.ip_out.ipa_type;
+	if (api.ip_out.ipa_type != AF_UNSPEC)
+		memcpy((uint8_t *)sockunion_get_addr(&lladdr), &api.ip_out.ip.addr,
```

Comment:
style warning?

Reply:
this line was above 80 byte column. I fixed it.